pandas/tests/io/test_parquet.py

```diff
@@ -691,6 +691,7 @@ def test_partition_cols_supported(self, pa, df_full):
             dataset = pq.ParquetDataset(path, validate_schema=False)
             assert len(dataset.partitions.partition_names) == 2
             assert dataset.partitions.partition_names == set(partition_cols)
+            assert read_parquet(path).shape == df.shape
```

Comment:
Making sure that `read_parquet` is more tested when reading from a directory.

The read DataFrame has shuffled rows. Is that expected?

Reply:
cc @jorisvandenbossche